### ISSUE TITLE
GGUF Q8_0 became much slower

### ISSUE THREAD

**Issue by DenkingOfficial** [lllyasviel/stable-diffusion-webui-forge#1502]
After 13d6f8ed900b0857e9872e67befb02f7ed54da35 commit speed became 4 times lower using flux1-dev-Q8_0.gguf and took 8 minutes for 1344x768 image, 20 steps, euler

On 424baa110de0e5e140b614b20070cb6703936e7f commit it works as intended: 2 minutes with same settings

**Comment by lllyasviel** [comment]
Can you provide full logs

**Comment by DenkingOfficial** [comment]
Yep, sure

Commit 424baa110de0e5e140b614b20070cb6703936e7f:

```
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Version: f2.0.1v1.10.1-previous-422-g424baa11
Commit hash: 424baa110de0e5e140b614b20070cb6703936e7f
Launching Web UI with arguments:
Total VRAM 8192 MB, total RAM 16282 MB
pytorch version: 2.3.1+cu121
Set vram state to: NORMAL_VRAM
Device: cuda:0 NVIDIA GeForce RTX 2060 SUPER : native
Hint: your device supports --cuda-malloc for potential speed improvements.
VAE dtype preferences: [torch.float32] -> torch.float32
CUDA Using Stream: False
C:\Software\SD\forge\system\python\lib\site-packages\transformers\utils\hub.py:127: FutureWarning: Using `TRANSFORMERS_CACHE` is deprecated and will be removed in v5 of Transformers. Use `HF_HOME` instead.
  warnings.warn(
Using pytorch cross attention
Using pytorch attention for VAE
ControlNet preprocessor location: C:\Software\SD\forge\webui\models\ControlNetPreprocessor
2024-08-26 23:22:32,730 - ControlNet - INFO - ControlNet UI callback registered.
Model selected: {'checkpoint_info': {'filename': 'C:\\Software\\SD\\forge\\webui\\models\\Stable-diffusion\\flux1-dev-Q8_0.gguf', 'hash': 'b44b9b8a'}, 'additional_modules': ['C:\\Software\\SD\\forge\\webui\\models\\VAE\\ae.safetensors', 'C:\\Software\\SD\\forge\\webui\\models\\text_encoder\\clip_l.safetensors', 'C:\\Software\\SD\\forge\\webui\\models\\text_encoder\\t5xxl_fp8_e4m3fn.safetensors'], 'unet_storage_dtype': None}
Using online LoRAs in FP16: True
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
Startup time: 22.6s (prepare environment: 4.6s, import torch: 10.1s, initialize shared: 0.3s, other imports: 0.7s, load scripts: 2.5s, create ui: 2.9s, gradio launch: 1.5s).
Environment vars changed: {'stream': False, 'inference_memory': 1024.0, 'pin_shared_memory': False}
[GPU Setting] You will use 87.50% GPU memory (7167.00 MB) to load weights, and use 12.50% GPU memory (1024.00 MB) to do matrix computation.
Loading Model: {'checkpoint_info': {'filename': 'C:\\Software\\SD\\forge\\webui\\models\\Stable-diffusion\\flux1-dev-Q8_0.gguf', 'hash': 'b44b9b8a'}, 'additional_modules': ['C:\\Software\\SD\\forge\\webui\\models\\VAE\\ae.safetensors', 'C:\\Software\\SD\\forge\\webui\\models\\text_encoder\\clip_l.safetensors', 'C:\\Software\\SD\\forge\\webui\\models\\text_encoder\\t5xxl_fp8_e4m3fn.safetensors'], 'unet_storage_dtype': None}
[Unload] Trying to free all memory for cuda:0 with 0 models keep loaded ...
StateDict Keys: {'transformer': 780, 'vae': 244, 'text_encoder': 196, 'text_encoder_2': 220, 'ignore': 0}
Using Detected T5 Data Type: torch.float8_e4m3fn
Using Detected UNet Type: gguf
Using pre-quant state dict!
GGUF state dict: {'F16': 476, 'Q8_0': 304}
Working with z of shape (1, 16, 32, 32) = 16384 dimensions.
K-Model Created: {'storage_dtype': 'gguf', 'computation_dtype': torch.float16}
Model loaded in 78.2s (unload existing model: 0.2s, forge model load: 78.0s).
[LORA] Loaded C:\Software\SD\forge\webui\models\Lora\zxs_v1_flux_lora_000006500.safetensors for KModel-UNet with 494 keys at weight 1.0 (skipped 0 keys)
Skipping unconditional conditioning when CFG = 1. Negative Prompts are ignored.
To load target model JointTextEncoder
Begin to load 1 model
[Unload] Trying to free 7723.54 MB for cuda:0 with 0 models keep loaded ...
[Memory Management] Current Free GPU Memory: 7140.00 MB
[Memory Management] Required Model Memory: 5153.49 MB
[Memory Management] Required Inference Memory: 1024.00 MB
[Memory Management] Estimated Remaining GPU Memory: 962.51 MB
Moving model(s) has taken 20.02 seconds
Distilled CFG Scale: 3.5
To load target model KModel
Begin to load 1 model
[Unload] Trying to free 17209.53 MB for cuda:0 with 0 models keep loaded ...
[Unload] Current free memory is 1865.43 MB ...
[Unload] Unload model JointTextEncoder
[Memory Management] Current Free GPU Memory: 7098.04 MB
[Memory Management] Required Model Memory: 12119.55 MB
[Memory Management] Required Inference Memory: 1024.00 MB
[Memory Management] Estimated Remaining GPU Memory: -6045.51 MB
Patching LoRA in on-the-fly.
Patching LoRAs for KModel: 100%|█████████████████████████████████████████████████████| 304/304 [01:11<00:00,  4.25it/s]
LoRA patching has taken 71.62 seconds
[Memory Management] Loaded to CPU Swap: 7326.02 MB (blocked method)
[Memory Management] Loaded to GPU: 4793.46 MB
Moving model(s) has taken 130.58 seconds
100%|██████████████████████████████████████████████████████████████████████████████████| 20/20 [01:58<00:00,  5.93s/it]
To load target model IntegratedAutoencoderKL███████████████████████████████████████████| 20/20 [01:48<00:00,  5.80s/it]
Begin to load 1 model
[Unload] Trying to free 8991.55 MB for cuda:0 with 0 models keep loaded ...
[Unload] Current free memory is 2274.36 MB ...
[Unload] Unload model KModel
[Memory Management] Current Free GPU Memory: 7094.07 MB
[Memory Management] Required Model Memory: 319.75 MB
[Memory Management] Required Inference Memory: 1024.00 MB
[Memory Management] Estimated Remaining GPU Memory: 5750.32 MB
Moving model(s) has taken 11.32 seconds
Total progress: 100%|██████████████████████████████████████████████████████████████████| 20/20 [02:02<00:00,  6.12s/it]
```

Commit e60bb1c96fbcc257a4dbfc8d212df24a363cf379:

```
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Version: f2.0.1v1.10.1-previous-426-ge60bb1c9
Commit hash: e60bb1c96fbcc257a4dbfc8d212df24a363cf379
Launching Web UI with arguments:
Total VRAM 8192 MB, total RAM 16282 MB
pytorch version: 2.3.1+cu121
Set vram state to: NORMAL_VRAM
Device: cuda:0 NVIDIA GeForce RTX 2060 SUPER : native
Hint: your device supports --cuda-malloc for potential speed improvements.
VAE dtype preferences: [torch.float32] -> torch.float32
CUDA Using Stream: False
C:\Software\SD\forge\system\python\lib\site-packages\transformers\utils\hub.py:127: FutureWarning: Using `TRANSFORMERS_CACHE` is deprecated and will be removed in v5 of Transformers. Use `HF_HOME` instead.
  warnings.warn(
Using pytorch cross attention
Using pytorch attention for VAE
ControlNet preprocessor location: C:\Software\SD\forge\webui\models\ControlNetPreprocessor
2024-08-26 23:38:21,835 - ControlNet - INFO - ControlNet UI callback registered.
Model selected: {'checkpoint_info': {'filename': 'C:\\Software\\SD\\forge\\webui\\models\\Stable-diffusion\\flux1-dev-Q8_0.gguf', 'hash': 'b44b9b8a'}, 'additional_modules': ['C:\\Software\\SD\\forge\\webui\\models\\VAE\\ae.safetensors', 'C:\\Software\\SD\\forge\\webui\\models\\text_encoder\\clip_l.safetensors', 'C:\\Software\\SD\\forge\\webui\\models\\text_encoder\\t5xxl_fp8_e4m3fn.safetensors'], 'unet_storage_dtype': None}
Using online LoRAs in FP16: True
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
Startup time: 22.4s (prepare environment: 5.0s, import torch: 9.6s, initialize shared: 0.3s, other imports: 0.8s, load scripts: 2.5s, create ui: 2.8s, gradio launch: 1.5s).
Environment vars changed: {'stream': False, 'inference_memory': 1024.0, 'pin_shared_memory': False}
[GPU Setting] You will use 87.50% GPU memory (7167.00 MB) to load weights, and use 12.50% GPU memory (1024.00 MB) to do matrix computation.
Model selected: {'checkpoint_info': {'filename': 'C:\\Software\\SD\\forge\\webui\\models\\Stable-diffusion\\flux1-dev-Q8_0.gguf', 'hash': 'b44b9b8a'}, 'additional_modules': ['C:\\Software\\SD\\forge\\webui\\models\\VAE\\ae.safetensors', 'C:\\Software\\SD\\forge\\webui\\models\\text_encoder\\clip_l.safetensors', 'C:\\Software\\SD\\forge\\webui\\models\\text_encoder\\t5xxl_fp8_e4m3fn.safetensors'], 'unet_storage_dtype': None}
Using online LoRAs in FP16: False
Model selected: {'checkpoint_info': {'filename': 'C:\\Software\\SD\\forge\\webui\\models\\Stable-diffusion\\flux1-dev-Q8_0.gguf', 'hash': 'b44b9b8a'}, 'additional_modules': ['C:\\Software\\SD\\forge\\webui\\models\\VAE\\ae.safetensors', 'C:\\Software\\SD\\forge\\webui\\models\\text_encoder\\clip_l.safetensors', 'C:\\Software\\SD\\forge\\webui\\models\\text_encoder\\t5xxl_fp8_e4m3fn.safetensors'], 'unet_storage_dtype': None}
Using online LoRAs in FP16: True
Loading Model: {'checkpoint_info': {'filename': 'C:\\Software\\SD\\forge\\webui\\models\\Stable-diffusion\\flux1-dev-Q8_0.gguf', 'hash': 'b44b9b8a'}, 'additional_modules': ['C:\\Software\\SD\\forge\\webui\\models\\VAE\\ae.safetensors', 'C:\\Software\\SD\\forge\\webui\\models\\text_encoder\\clip_l.safetensors', 'C:\\Software\\SD\\forge\\webui\\models\\text_encoder\\t5xxl_fp8_e4m3fn.safetensors'], 'unet_storage_dtype': None}
[Unload] Trying to free all memory for cuda:0 with 0 models keep loaded ...
StateDict Keys: {'transformer': 780, 'vae': 244, 'text_encoder': 196, 'text_encoder_2': 220, 'ignore': 0}
Using Detected T5 Data Type: torch.float8_e4m3fn
Using Detected UNet Type: gguf
Using pre-quant state dict!
GGUF state dict: {'F16': 476, 'Q8_0': 304}
GGUF backed 304 layers.
Working with z of shape (1, 16, 32, 32) = 16384 dimensions.
K-Model Created: {'storage_dtype': 'gguf', 'computation_dtype': torch.float16}
Model loaded in 74.7s (unload existing model: 0.2s, forge model load: 74.5s).
[LORA] Loaded C:\Software\SD\forge\webui\models\Lora\zxs_v1_flux_lora_000006500.safetensors for KModel-UNet with 494 keys at weight 1.0 (skipped 0 keys)
Skipping unconditional conditioning when CFG = 1. Negative Prompts are ignored.
To load target model JointTextEncoder
Begin to load 1 model
[Unload] Trying to free 7723.54 MB for cuda:0 with 0 models keep loaded ...
[Memory Management] Current Free GPU Memory: 7140.00 MB
[Memory Management] Required Model Memory: 5153.49 MB
[Memory Management] Required Inference Memory: 1024.00 MB
[Memory Management] Estimated Remaining GPU Memory: 962.51 MB
Moving model(s) has taken 17.53 seconds
Distilled CFG Scale: 3.5
To load target model KModel
Begin to load 1 model
[Unload] Trying to free 17209.53 MB for cuda:0 with 0 models keep loaded ...
[Unload] Current free memory is 1865.43 MB ...
[Unload] Unload model JointTextEncoder
[Memory Management] Current Free GPU Memory: 7098.04 MB
[Memory Management] Required Model Memory: 12119.55 MB
[Memory Management] Required Inference Memory: 1024.00 MB
[Memory Management] Estimated Remaining GPU Memory: -6045.51 MB
Patching LoRA in on-the-fly.
Patching LoRAs for KModel: 100%|█████████████████████████████████████████████████████| 304/304 [01:10<00:00,  4.29it/s]
LoRA patching has taken 70.97 seconds
[Memory Management] Loaded to CPU Swap: 7326.02 MB (blocked method)
[Memory Management] Loaded to GPU: 4793.46 MB
Moving model(s) has taken 120.47 seconds
100%|██████████████████████████████████████████████████████████████████████████████████| 20/20 [10:57<00:00, 32.89s/it]
To load target model IntegratedAutoencoderKL███████████████████████████████████████████| 20/20 [10:17<00:00, 34.56s/it]
Begin to load 1 model
[Unload] Trying to free 8991.55 MB for cuda:0 with 0 models keep loaded ...
[Unload] Current free memory is 2299.30 MB ...
[Unload] Unload model KModel
Press any key to continue . . .
```

On e60bb1c96fbcc257a4dbfc8d212df24a363cf379 it didn't even finished a generation, process just turned off

Same speed issue with the latest commit too

**Comment by DenkingOfficial** [comment]
Image:

![00244-1550452902](https://github.com/user-attachments/assets/d0b596ec-2fca-44e5-9828-95018be27faa)

And settings used for reproduction:

PS1 game graphics style, PS2 game graphics style, very low poly, low quality textures, a pink sunset cityscape with a row of apartment buildings on the left side, clouds are scattered across the sky, the buildings in the background are tall and have multiple floors and balconies, overall mood of the photo is peaceful and serene <lora:zxs_v1_flux_lora:1> Steps: 20, Sampler: Euler, Schedule type: Simple, CFG scale: 1, Distilled CFG Scale: 3.5, Seed: 1550452902, Size: 1344x768, Model hash: 52cfce60d7, Model: flux1-dev-Q8_0, Lora hashes: "zxs_v1_flux_lora: e3b0c44298fc", Version: f2.0.1v1.10.1-previous-422-g424baa11, Diffusion in Low Bits: Automatic (fp16 LoRA), Module 1: ae, Module 2: clip_l, Module 3: t5xxl_fp8_e4m3fn

**Comment by lllyasviel** [comment]
something went wrong severely in both logs.
patching lora on the fly should not take any time but shows 70 seconds in both logs.
what will happen if you set gpu weight to 0

**Comment by DenkingOfficial** [comment]
> what will happen if you set gpu weight to 0

Here're my settings

![image](https://github.com/user-attachments/assets/8dc2b60b-9c93-4ab2-a3ff-a3dfdfc33c24)

> patching lora on the fly should not take any time but shows 70 seconds in both logs.

I updated to the latest commit, but still

![image](https://github.com/user-attachments/assets/2c5ca4ae-0524-4421-8c1f-3c4396b605a4)

Also, it just crashed after 100%

**Comment by lllyasviel** [comment]
Do the following things:

1. Set gpu weight to 0. Run again. Give me screenshot of **both** your console with speed and Task Manager GPU section
2. Set gpu weight to 7167. Run again. Give me screenshot of **both** your console with speed and Task Manager GPU section

Task Manager GPU section is this:


![image](https://github.com/user-attachments/assets/5d5f37b1-4db5-49e6-bec1-a5d03343aebd)

**Comment by lllyasviel** [comment]
Also, I put a hotfix just now, update and see if it makes any difference

**Comment by ALBIHANY** [comment]
@lllyasviel  Q8 stopped working for me for sometimes too and i reported that on some of your commit here is my details just now after updating to the new commit you added 

![image](https://github.com/user-attachments/assets/0770cb06-8a93-4a81-8c09-ad50c05f327c)

![image](https://github.com/user-attachments/assets/cdcdc100-0c24-4084-8268-f35a54d1336e)


This model used to work just fine on forge now it doesn't work at all, I can run it normally on comfy

**Comment by DenkingOfficial** [comment]
> Set gpu weight to 0. Run again. Give me screenshot of **both** your console with speed and Task Manager GPU section

![image](https://github.com/user-attachments/assets/a72bc10b-4deb-4860-940e-c9bb9835f6e8)

> Set gpu weight to 7167. Run again. Give me screenshot of **both** your console with speed and Task Manager GPU section

![image](https://github.com/user-attachments/assets/85126cea-1268-4dc3-aed9-328c4954e5cd)

> Also, I put a hotfix just now, update and see if it makes any difference

Nope, testing on 25662974f8ff19a8be8bc013f305cfe3b7b99a8f, still slow

On 424baa110de0e5e140b614b20070cb6703936e7f it's blazingly fast compared to later commits

![image](https://github.com/user-attachments/assets/d16193a8-687c-4588-a686-d5094e223e50)

**Comment by anon0730** [comment]
Looks like something is causing more system RAM to be used on later commits while on the earlier commit only about 10 GB is used during the actual generation. The way your RAM is completely used up suggests things are being pushed into swap (paging file) which can be very slow. 

This is off topic and not a solution to why this is happening but consider purchasing more system RAM if at all possible. I upgraded to 32 GB and it makes a huge difference in general when loading and unloading various modules around.

**Comment by DenkingOfficial** [comment]
> The way your RAM is completely used up suggests things are being pushed into swap (paging file) which can be very slow.

Yep, I also noticed that

> This is off topic and not a solution to why this is happening but consider purchasing more system RAM if at all possible. I upgraded from 16 GB to 32 GB and it makes a huge difference in general when loading and unloading various modules around.

Agree, my setup is very far from "Ok" tier, but still usable

**Comment by DenkingOfficial** [comment]
> I made second attempt to fix this. update and test again

Now it works like a charm! Thanks!
Still prepatches a LoRA, but it is unrelated to this issue I guess

![image](https://github.com/user-attachments/assets/7b26bc53-e3e8-4f52-92e3-0627afec2598)

> Thank you very much for your cooperation.

Thank you for maintaining such an amazing piece of software!

**Comment by lllyasviel** [comment]
cool
